unit/goto-instrument/cover_instrument.cpp

```diff
@@ -0,0 +1,20 @@
+#include <goto-instrument/cover_instrument.h>
+#include <testing-utils/use_catch.h>
+
+TEST_CASE("cover_intrument_end_of_function")
+{
+  // Arrange
+  goto_programt goto_program{};
+  goto_program.add(
+    goto_programt::make_function_call(code_function_callt({}, {})));
+  goto_program.add(
+    goto_programt::make_function_call(code_function_callt({}, {})));
+  goto_program.add(goto_programt::make_return());
+  // Act
+  cover_instrument_end_of_function("foo", goto_program);
+  // Assert
+  REQUIRE(goto_program.instructions.size() == 4);
+  const auto newly_inserted = std::next(goto_program.instructions.begin(), 2);
```

Comment:
:pick: Personally would find it easier to see that this is being inserted after the second function call phrased like this:

```
const auto newly_inserted = goto_program.instructions[2]
```

though I can see this is the same.

Reply:
can't do, it's a freaking `std::list` no index operator here :cry: